.github/workflows/pr-tests.yaml

```diff
@@ -0,0 +1,42 @@
+name: PR Tests
+
+on:
+  pull_request: {}
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.16'
+      - name: Cache go-build and mod
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build/
+            ~/go/pkg/mod/
+          key: ${{ hashFiles('go.sum') }}
+      - name: Setup Kind CLI
+        run: |
+          sudo curl -sSLo /usr/local/bin/kind https://github.com/kubernetes-sigs/kind/releases/download/v0.10.0/kind-linux-amd64
+          sudo chmod +x /usr/local/bin/kind
+      - name: Setup Kubectl
+        uses: azure/setup-kubectl@v1
+        id: install
+      - name: Setup Flux v2 CLI
+        uses: fluxcd/flux2/action@main
+      - name: Setup Ginkgo
+        run: |
+          go get -u github.com/onsi/ginkgo/ginkgo
+      - name: Setup lint
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.41.0
+      - name: Test
+        run: |
+          SKIP_CREDENTIAL_TESTS=true PATH=$PATH:$(go env GOPATH)/bin make test
```

Comment:
new env var for skipping credentials test. `make test` runs `make lint`, which I like. so I've merged the lint workflow into this workflow